modal/stub.py

```diff
@@ -462,11 +463,12 @@ def _add_function(self, function: _Function) -> _FunctionHandle:
         if function.tag in self._blueprint:
             old_function = self._blueprint[function.tag]
             if isinstance(old_function, _Function):
-                logger.warning(
-                    f"Warning: Tag {function.tag} collision!"
-                    f" Overriding existing function [{old_function._info.module_name}].{old_function._info.function_name}"
-                    f" with new function [{function._info.module_name}].{function._info.function_name}"
-                )
+                if not is_notebook():
```

Comment:
oh because you want to redefine functions?

Reply:
wonder if it makes sense to do something like @aksh-at mentioned where we have a helper function `stub.notebook()` or whatever which creates a background task that always syncs the stub with the running app

Reply:
Yeah, but those are two slightly unrelated problems. Redefining functions, even without having them repopulate a running app, is pretty common in notebooks as you iterate on the function definitions (and don't want to restart the interpreter/kernel every time you save the cell), so I think it makes sense to ignore these deprecations in notebook contexts.

Adding something like `stub.notebook()` (probably part of a notebook magic?) would be helpful too, in case the app has already been constructed and is live when the redefinition happens.